lib/ansible/modules/remote_management/manageiq/manageiq_group.py

```diff
@@ -0,0 +1,601 @@
+            resource['description'] = description
+            changed = True
+
+        if tenant and group['tenant_id'] != tenant['id']:
+            resource['tenant'] = dict(id=tenant['id'])
+            changed = True
+
+        if role and current_role != role['name']:
+            resource['role'] = dict(id=role['id'])
+            changed = True
+
+        if managed_filters or belongsto_filters:
+
+            # Only compare if filters are supplied
+            entitlement = group['entitlement']
+
+            if 'filters' not in entitlement:
+                # No existing filters exist, use supplied filters
+                resource['filters'] = {'managed': managed_filters, "belongsto": belongsto_filters}
+                changed = True
+            else:
+                current_filters = entitlement['filters']
+                new_filters = self.edit_group_edit_filters(current_filters,
+                                                           managed_filters, managed_filters_merge_mode,
+                                                           belongsto_filters, belongsto_filters_merge_mode)
+                if new_filters:
+                    resource['filters'] = new_filters
+                    changed = True
+
+        if not changed:
+            return dict(
+                changed=False,
+                msg="group %s is not changed." % group['description'])
+
+        # try to update group
+        try:
+            self.client.post(group['href'], action='edit', resource=resource)
+            changed = True
+        except Exception as e:
+            self.module.fail_json(msg="failed to update group %s: %s" % (group['name'], str(e)))
+
+        return dict(
+            changed=changed,
+            msg="successfully updated the group %s with id %s" % (group['description'], group['id']))
+
+    def edit_group_edit_filters(self, current_filters, managed_filters, managed_filters_merge_mode,
+                                belongsto_filters, belongsto_filters_merge_mode):
+        """ Edit a manageiq group filters.
+
+        Returns:
+            None if no the group was not updated
+            If the group was updated the post body part for updating the group
+        """
+        filters_updated = False
+        new_filters_resource = {}
+
+        # Process belongsto filters
+        if 'belongsto' in current_filters:
+            current_belongsto_set = set(current_filters['belongsto'])
+        else:
+            current_belongsto_set = set()
+
+        if belongsto_filters:
+            new_belongsto_set = set(belongsto_filters)
+        else:
+            new_belongsto_set = set()
+
+        if current_belongsto_set == new_belongsto_set:
+            new_filters_resource['belongsto'] = current_filters['belongsto']
+        else:
+            if belongsto_filters_merge_mode == 'merge':
+                current_belongsto_set.update(new_belongsto_set)
```

Comment:
Is this deliberately different from `merge_values_nested_array(current_managed, managed_filters)` below?
Ouch, I see, belongsto is 1 level while managed is 2 levels:
```
    "filters" : {
      "belongsto" : [ "/managed/area/1", "/managed/area/2", "/managed/area/3" ],
      "managed" : [[ "/managed/infra/1", "/managed/infra/2"], ["/managed/other/3"]]
    }
```
LGTM :+1:

Reply:
Yep the belongs to is easy to merge...